group-coordinator/src/main/java/org/apache/kafka/coordinator/group/share/ShareCoordinatorService.java

```diff
@@ -333,9 +337,137 @@ public CompletableFuture<WriteShareGroupStateResponseData> writeState(RequestCon
     });
   }
 
+  private ReadShareGroupStateResponseData generateErrorReadStateResponse(ReadShareGroupStateRequestData request, Errors error, String errorMessage) {
+    ReadShareGroupStateResponseData responseData;
+    if (!isEmpty(request.topics())) {
+      responseData = new ReadShareGroupStateResponseData()
+          .setResults(request.topics().stream()
+              .map(topicData -> {
+                ReadShareGroupStateResponseData.ReadStateResult resultData = new ReadShareGroupStateResponseData.ReadStateResult();
+                resultData.setTopicId(topicData.topicId());
+                if (!isEmpty(topicData.partitions())) {
+                  resultData.setPartitions(topicData.partitions().stream()
+                      .map(partitionData -> ReadShareGroupStateResponse.getErrorResponsePartitionResult(
+                          partitionData.partition(), error, errorMessage
+                      )).collect(Collectors.toList()));
+                } else {
+                  resultData.setPartitions(Collections.singletonList(ReadShareGroupStateResponse.getErrorResponsePartitionResult(
+                      -1, error, errorMessage
+                  )));
+                }
+                return resultData;
+              }).collect(Collectors.toList()));
+    } else {
+      responseData = ReadShareGroupStateResponse.getErrorResponseData(
```

Comment:
This is the case where there were no topics in the request I think, but you're building a response with a single topic-partition (ZERO_UUID and -1). This doesn't seem right.